trax/tf_numpy/examples/mnist/train.py

```diff
@@ -18,71 +18,57 @@
-  """Runs the training."""
-  print('Loading data')
-  training_data, validation_data, test_data = dataset.load()
-  print('Loaded dataset with {} training, {} validation and {} test examples.'.
-        format(
-            len(training_data[0]), len(validation_data[0]), len(test_data[0])))
-
-  assert len(training_data[0]) % batch_size == 0
-  assert len(validation_data[0]) % batch_size == 0
-  assert len(test_data[0]) % batch_size == 0
-
-  def build_iterator(data, infinite=True):
-    """Build the iterator for inputs."""
-    index = 0
-    size = len(data[0])
-    while True:
-      if index + batch_size > size:
-        if infinite:
-          index = 0
-        else:
-          return
-      yield data[0][index:index + batch_size], data[1][index:index + batch_size]
-      index += batch_size
-
-  train_iter = build_iterator(training_data)
-  model = model_lib.Model([30])
-
-  for i in range(num_training_iters):
-    train_x, train_y = next(train_iter)
-    model.train(train_x, train_y, learning_rate)
-    if not (i + 1) % validation_steps:
-      validation_iter = build_iterator(validation_data, infinite=False)
-      correct_predictions = 0
-      for valid_x, valid_y in validation_iter:
-        correct_predictions += model.evaluate(valid_x, valid_y)
-      print('{}/{} correct validation predictions.'.format(
-          correct_predictions, len(validation_data[0])))
-
-
-def main(unused_argv):
-  train(FLAGS.batch_size, FLAGS.learning_rate, FLAGS.num_training_iters,
-        FLAGS.validation_steps)
+    """ training loop """
+
+    mnist_dataset = MNIST(batch_size)
+    train_iter = mnist_dataset.iterator('train', infinite=True)
+
+    model = Model([512, 256, 128], learning_rate=LEARNING_RATE)
+
+    for i in range(num_training_iters):
+
+            train_x, train_y = next(train_iter)
+            loss = model.train(train_x, train_y)
+
+            # Calculate and print the train and test accuracy
+            if not (i + 1) % validation_steps:
+                training_iter = mnist_dataset.iterator('train', infinite=False)
+                validation_iter = mnist_dataset.iterator('test', infinite=False)
+
+                correct_train_predictions = 0
+                for train_x, train_y in training_iter:
+                    correct_train_predictions += model.evaluate(train_x, train_y)
+
+                correct_val_predictions = 0
+                for valid_x, valid_y in validation_iter:
+                    correct_val_predictions += model.evaluate(valid_x, valid_y)
+
+
+                print('iter {}: {} training accuracy, {} test accuracy.'.format(
+                    i+1,
+                    round(correct_train_predictions/mnist_dataset.train_len, 4),
+                    round(correct_val_predictions/mnist_dataset.val_len, 4)))
 
 
 if __name__ == '__main__':
   tf.compat.v1.enable_eager_execution()
-  app.run()
+  train(BATCH_SIZE, LEARNING_RATE, NUM_TRAINING_ITERS, VALIDATION_STEPS)
```

Comment:
Let's change it back to app.run with a "def main". This is important for this to work reasonably internally.